### PR TITLE
Meter fixed-size UInt* values

### PR DESCRIPTION
**Pull request by turbolent**
Work towards https://github.com/dapperlabs/cadence-private-issues/issues/2
______

- [ ] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by SupunS — APPROVED** [review]

File: runtime/interpreter/interpreter_expression.go [hunk 1]
```diff
@@ -512,17 +512,23 @@ func (interpreter *Interpreter) NewIntegerValueFromBigInt(value *big.Int, intege
 
 	// UInt*
 	case sema.UInt8Type:
-		return UInt8Value(value.Int64())
+		common.UseMemory(memoryGauge, Uint8MemoryUsage)
+		return NewUnmeteredUInt8Value(uint8(value.Uint64()))
```
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/interpreter/interpreter_expression.go:516]: Great question! Theoretically we could, but I was worried that in this hot-code path using the "convenience" constructor was maybe inefficient, so I basically "inlined" / manually metered. It is probably a premature optimization.

What do you think?

[the remark this answers is not in the document]
Comment (SupunS) [reply runtime/interpreter/interpreter_expression.go:516]: yeah, closures in hot paths can have a perf hit at the end of the day (e.g: https://github.com/onflow/cadence/issues/1451). I wholeheartedly agree to remove closures as much as we could. I guess this is one of those cases where we have to pick from 'perf' vs 'low potential for mistakes'.

Another option is to introduce a set of constructors that takes an `*big.Int`. But that would be simply moving this code to a different location, and might not be useful unless it's re-used. So I'm good with either way 👍

File: runtime/interpreter/value.go [hunk 2]
```diff
@@ -8868,12 +9175,31 @@ type UInt128Value struct {
...
+	)
+}
+
+func NewUnmeteredUInt128ValueFromUint64(value uint64) UInt128Value {
+	return NewUnmeteredUInt128ValueFromBigInt(new(big.Int).SetUint64(value))
 }
 
-func NewUInt128ValueFromBigInt(value *big.Int) UInt128Value {
-	return UInt128Value{BigInt: value}
+var Uint128MemoryUsage = common.NewNumberMemoryUsage(16)
```
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/interpreter/value.go]: Right, I guess we might meter more for bigint-backed types, so we should meter them as that. Changed in bc40b503cd88138d743a2fd34b7bf703cd492684

File: runtime/interpreter/interpreter_expression.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (turbolent) [reply runtime/interpreter/interpreter_expression.go:502]: Improved in 1ed200a99c47ec281e1fb9ab1c4c3e900fc882e2

**Pull request merged**